analysis/test/src/pointers.rs

```diff
@@ -540,6 +589,73 @@ pub unsafe extern "C" fn insertion_sort(n: libc::c_int, p: *mut libc::c_int) {
         i += 1
     }
 }
+
+/*
+    This is a minimal breaking example from lighttpd where instrumented code failed
+    to compile.
+*/
+#[no_mangle]
+#[cold]
+unsafe extern "C" fn log_error_va_list_impl(
+    mut errh: *const libc::c_void,
+    filename: *const libc::c_char,
+    line: libc::c_uint,
+    fmt: *const libc::c_char,
+    mut ap: ::std::ffi::VaList,
+    perr: libc::c_int,
+) {
+    let mut prefix: [libc::c_char; 40] = [0; 40];
+    vsprintf(prefix.as_mut_ptr(), fmt, ap.as_va_list());
+}
+pub unsafe extern "C" fn log_error(
+    errh: *mut libc::c_void,
+    filename: *const libc::c_char,
+    line: libc::c_uint,
+    mut fmt: *const libc::c_char,
+    mut args: ...
+) {
```

Comment:
Is a variadic function necessary?  `miri` can't run variadic functions, so if we don't need to test variadic functions, we shouldn't add them.  I don't want UB creeping back into this file.

If we do want to test variadics themselves, we could do that separately (where we're more sure there's no UB by human inspection), and have a `cfg!(feature = "miri")` around calling it.  If necessary for this test, we could also use that strategy here.

Reply:
this doesn't run / get called anyway, and yes it was necessary to reproduce the instrumentation error I encountered

Reply:
Okay, if it's not actually run, then it's fine with `miri`.